### ISSUE TITLE
Croner returns different values in case of trailing whitespace

### ISSUE THREAD

**Issue by haggholm** [Hexagon/croner#215]
**Describe the bug**

If a cron expression has trailing whitespace, Croner appears to ‘hallucinate’ an extra (empty?) field. At a first guess I'd suggest that maybe it chooses mode depending on the number of results from something like string.split().

**To Reproduce**

```javascript
require("croner")("0 0 12 9 *").nextRuns(3);
```
→
```
[
  2024-09-12T07:01:00.000Z,
  2025-09-12T07:01:00.000Z,
  2026-09-12T07:01:00.000Z
]
```

But
```javascript
require("croner")("0 0 12 9 * ").nextRuns(3);
```
→
```
[
  2023-11-09T20:00:00.000Z,
  2023-12-09T20:00:00.000Z,
  2024-01-09T20:00:00.000Z
]
```

**Expected behavior**

I'd expect it to _either_ trim the input expression and ignore extraneous whitespace, or failing that, throw an error. It's very surprising to get different results based on a trailing space.

**System:**
 - OS: Linux
 - Runtime: Node.js 16
 - Croner version: 7.0.0

**Additional context**

**Comment by Hexagon** [comment]
Issue confirmed. 

When using a trailing space together with a five part pattern, it's treated as a six part pattern with an empty instruction for weekdays - which interestingly enough is accepted and treated as a wildcard. I will figure out a good solution to this, along with some additional tests.

**Comment by Hexagon** [comment]
This issue is resolved in `croner@dev` or `croner@7.0.3-dev.0` by throwing on empty parts (which in this case is caused by a trailing space).
